packages/doenetml-worker-rust/lib-doenetml-core/src/core/math_via_wasm.rs

```diff
@@ -0,0 +1,327 @@
+    ))
+}
+
+/// Parsing a string into math using the `math-expressions` text parser.
+///
+/// Parameters:
+/// - `text`: the string to be parsed
+/// - `split_symbols`: if true, split multi-character symbols that do not contain an number
+///   into the product of the characters.
+/// - `function_symbols`: a list of the symbols that will be treated as a function,
+///   i.e., one of these symbols followed by arguments in parentheses
+///   will be interpreted as apply that function to the arguments (rather than multiplication)
+///
+/// Examples:
+/// - `parse_text_into_math("xy", true, &["f"])` is `x` times `y`.
+/// - `parse_text_into_math("xy", false, &["f"])` is the multi-character symbol `xy`.
+/// - `parse_text_into_math("g(x)", true, &["f", "g"])` is the function `g` evaluated at `x`.
+/// - `parse_text_into_math("g(x)", true, &["f"])` is `g` times `x`.
+#[cfg(all(not(feature = "testing"), feature = "web"))]
+pub fn parse_text_into_math<Text: AsRef<str>, FnSymbol: AsRef<str>>(
+    text: Text,
+    split_symbols: bool,
+    function_symbols: &[FnSymbol],
+) -> Result<String, anyhow::Error> {
+    let js_function_symbols = JsValue::from(
+        function_symbols
+            .iter()
+            .map(|x| JsValue::from_str(x.as_ref()))
+            .collect::<Array>(),
+    );
+    let result: JsString = parseTextIntoMath(
+        JsString::from(text.as_ref()),
+        Boolean::from(split_symbols),
+        js_function_symbols,
+    )
+    .map_err(|e| anyhow!("{:?}", e))?;
+    Ok(result.into())
+}
+#[cfg(any(feature = "testing", not(feature = "web")))]
+pub fn parse_text_into_math<Text: AsRef<str>, FnSymbol: AsRef<str>>(
+    _text: Text,
+    _split_symbols: bool,
+    _function_symbols: &[FnSymbol],
+) -> Result<String, anyhow::Error> {
+    Err(anyhow!(
+        "parse_text_into_math is only available when compiled with the `web` feature".to_string()
+    ))
+}
+
+/// Parsing a string into math using the `math-expressions` latex parser.
+///
+/// Parameters:
+/// - `latex`: the string to be parsed
+/// - `split_symbols`: if true, split multi-character symbols that do not contain an number
+///   into the product of the characters.
+/// - `function_symbols`: a list of the symbols that will be treated as a function,
+///   i.e., one of these symbols followed by arguments in parentheses
+///   will be interpreted as apply that function to the arguments (rather than multiplication)
+///
+/// Examples:
+/// - `parse_latex_into_math(r#"\frac{xy}{z}"#, true, &["g"])` is `x` times `y` divided by `z`.
+/// - `parse_latex_into_math(r#"\frac{xy}{z}"#, false, &["g"])` is the multi-character symbol `xy` divided by `z`.
+/// - `parse_latex_into_math("g(x)", true, &["f", "g"])` is the function `g` evaluated at `x`.
+/// - `parse_latex_into_math("g(x)", true, &["f"])` is `g` times `x`.
+#[cfg(all(not(feature = "testing"), feature = "web"))]
+pub fn parse_latex_into_math<Latex: AsRef<str>, FnSymbol: AsRef<str>>(
```

Comment:
This function is `str -> str`, so it's a little weird to have `parse_` in its name since we don't get the parse tree as a result. `latex_to_math` perhaps?

Reply:
That's an excellent point. There should be an opaque type `struct JsMathExpr(String)` or `struct MathObject(String)` or something like that. Then this type can be passed around and we don't have to know what's inside unless we are directly manipulating it.